tdigest/accumulation_tree.py

```diff
@@ -0,0 +1,151 @@
+import operator
+from bintrees.rbtree import RBTree, Node
+
+
+class AccumulationTree(RBTree):
+    def __init__(self, mapper, reducer=operator.add, zero=0):
+        RBTree.__init__(self)
+        self._zero = zero
+        self._mapper = mapper
+        self._reducer = reducer
+
+        self._dirty_nodes = set()
+
+        class AccumulatingNode(Node):
```

Comment:
Haha yes, ever since i learned about it, I've been able to cut a lot of

    def __init__(self, variable1, variable2, ...)
        self._variable1 = variable1
        self._variable2 = variable2
        ....
boilerplate.